src/main/webapp/pipe.js

```diff
@@ -311,124 +283,150 @@ function pipelineUtils() {
     }
 }
 
+function addPipelineHeader(html, component, data, c, resURL) {
+    html.push('<h1>' + htmlEncode(component.name));
+    if (data.allowPipelineStart) {
+        if (component.workflowComponent) {
+            html.push('&nbsp;<a id="startpipeline-' + c  +'" class="task-icon-link" href="#" onclick="triggerBuild(\'' + component.workflowUrl + '\', \'' + data.name + '\')">');
+        } else if (component.firstJobParameterized) {
+            html.push('&nbsp;<a id="startpipeline-' + c  +'" class="task-icon-link" href="#" onclick="triggerParameterizedBuild(\'' + component.firstJobUrl + '\', \'' + data.name + '\')">');
+        } else {
+            html.push('&nbsp;<a id="startpipeline-' + c  +'" class="task-icon-link" href="#" onclick="triggerBuild(\'' + component.firstJobUrl + '\', \'' + data.name + '\')">');
+        }
+        html.push('<img class="icon-clock icon-md" title="Build now" src="' + resURL + '/images/24x24/clock.png">');
+        html.push('</a>');
+    }
+    html.push('</h1>');
+}
+
+function checkDataForError(data, errrorDivId) {
+    var cErrorDiv = Q('#' + errrorDivId);
+    if (data.error) {
+        cErrorDiv.html('Error: ' + data.error).show();
+    } else {
+        cErrorDiv.hide().html('');
+    }
+}
+
+function isTaskLoggedToConsole(task) {
+    return task.status.success
+        || task.status.failed
+        || task.status.unstable
+        || task.status.cancelled;
+}
+
 function getPagination(showAvatars, component) {
     var html = [];
-    if (!showAvatars) {
-       if (component.pagingData != '') {
-           html.push("<div class='pagination'>");
-           html.push(component.pagingData);
-           html.push("</div>");
-       }
+    if (!showAvatars && component.pagingData != '') {
+        html.push('<div class="pagination">');
+        html.push(component.pagingData);
+        html.push('</div>');
     }
-    return html.join("");
+    return html.join('');
 }
 
 function getLink(data, link) {
-    if (data.linkRelative) {
-        return link;
-    } else {
-        return rootURL + "/" + link;
-    }
+    return data.linkRelative
```

Comment:
What is the reasoning behind using the ? : syntax in favor of an if / else here?